Sources/Commands/GenerateShellScript.swift

```diff
@@ -0,0 +1,328 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Foundation
+import Basic
+import Utility
+
+// todo: support positional arguments; eg generate-shell-script [shell-name]
+
+func generateShellScript(forShell shell: Shell, print: (String) -> ()) {
+    let root = ArgumentParser(usage: "", overview: "Swift compiler")
+    root.add(subparser: "build", parser: SwiftBuildTool(args: []).parser)
+    root.add(subparser: "package", parser: SwiftPackageTool(args: []).parser)
+    root.add(subparser: "test", parser: SwiftTestTool(args: []).parser)
+
+    // Swift compiler flags
+    _ = root.add(option: "-assert-config", kind: String.self, usage: "Specify the assert_configuration replacement. Possible values are Debug, Release, Unchecked, DisableReplacement.")
```

Comment:
We currently [already do that][0]. So my idea is to either generate them (like in this PR), or include them as static code. I would rather not lose the ability to complete compiler arguments.

[0]: https://github.com/apple/swift-package-manager/blob/914931d932e63ddeeaa1ddb59f4b8ca42a125e5e/Utilities/zsh/_swift#L122-L161